go/buildhelper/node.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/helpertypes"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+var (
+	nonceLock sync.RWMutex
+)
+
+type EthNode struct {
+	port      uint
+	ipaddress string
+	apiClient *EthAPI
+	id        common.Address
+}
+
+func NewEthNode(id common.Address, ipaddress string, port uint) (obscurocommon.L1Node, error) {
+	apiClient := NewEthAPI(ipaddress, port)
+	err := apiClient.Connect()
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to the eth node - %w", err)
+	}
+
+	log.Log(fmt.Sprintf("Initializing eth node at contract: %s", buildconstants.CONTRACT_ADDRESS))
+	return &EthNode{
+		ipaddress: ipaddress,
+		port:      port,
+		apiClient: apiClient,
+		id:        id,
+	}, nil
+}
+
+func (e *EthNode) RPCBlockchainFeed() []*types.Block {
+	var availBlocks []*types.Block
+
+	block, err := e.apiClient.ApiClient.BlockByNumber(context.Background(), nil)
+	if err != nil {
+		panic(err)
+	}
+	availBlocks = append(availBlocks, block)
+
+	for {
+		parentHash := block.ParentHash()
+		// todo set this to genesis hash
+		if parentHash.Hex() == "0x0000000000000000000000000000000000000000000000000000000000000000" {
+			break
+		}
+
+		block, err = e.apiClient.ApiClient.BlockByHash(context.Background(), block.ParentHash())
+		if err != nil {
+			fmt.Printf("ERROR %v\n", err)
+		}
+
+		availBlocks = append(availBlocks, block)
+	}
+
+	// todo double check the list is ordered [genesis, 1, 2, 3, 4, ..., last]
+	for i, j := 0, len(availBlocks)-1; i < j; i, j = i+1, j-1 {
+		availBlocks[i], availBlocks[j] = availBlocks[j], availBlocks[i]
+	}
+	return availBlocks
+}
+
+func (e *EthNode) BroadcastTx(t obscurocommon.EncodedL1Tx) {
+	nonceLock.Lock()
+	defer nonceLock.Unlock()
+
+	privateKey := Addr1PK()
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		panic("error casting public key to ECDSA")
+	}
+
+	fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+	nonce, err := e.apiClient.ApiClient.PendingNonceAt(context.Background(), fromAddress)
+	if err != nil {
+		panic(err)
+	}
+
+	l1tx, err := t.Decode()
+	if err != nil {
+		panic(err)
+	}
+
+	l1txData := obscurocommon.TxData(&l1tx)
+
+	ethTx := &types.LegacyTx{
+		Nonce:    nonce,
+		GasPrice: big.NewInt(20000000000),
+		Gas:      1024_000_000,
+		To:       &buildconstants.CONTRACT_ADDRESS,
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(buildconstants.CONTRACT_ABI))
+	if err != nil {
+		panic(err)
+	}
+
+	switch l1txData.TxType {
+	case obscurocommon.DepositTx:
+		ethTx.Value = big.NewInt(int64(l1txData.Amount))
```

Comment:
each of these cases should be a function:
`func createRollupTx()`
`func createDepositTx()`
etc

And then eventually, these functions would be called directly, when we get rid of our special format. (we'll have to change the mock thing as well for that)

Reply:
I'll capture this on a todo 👍